webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/RudderJsonRequest.scala

```diff
@@ -0,0 +1,40 @@
+package com.normation.rudder.rest
+
+import com.normation.errors.PureResult
+import com.normation.errors.Unexpected
+import net.liftweb.common.EmptyBox
+import net.liftweb.common.Full
+import net.liftweb.http.Req
+import zio.json._
+
+object RudderJsonRequest {
+
+  /**
+   * Parse request body as JSON, and decode it as type `A`.
+   * This is the root method to transform a JSON query into a Rest object.
+   */
+  def parseJson[A](req: Req)(implicit decoder: JsonDecoder[A]): PureResult[A] = {
+    if (req.json_?) {
+      // copied from `Req.forcedBodyAsJson`
+      def r  = """; *charset=(.*)""".r
+      def r2 = """[^=]*$""".r
+      def charset: String = req.contentType.flatMap(ct => r.findFirstIn(ct).flatMap(r2.findFirstIn)).getOrElse("UTF-8")
+      // end copy
+
+      req.body match {
+        case eb: EmptyBox => Left(Unexpected((eb ?~! "error when accessing request body").messageChain))
+        case Full(bytes) => decoder.decodeJson(new String(bytes, charset)).left.map(Unexpected(_))
+      }
+    } else {
+      Left(Unexpected("Cannot parse non-JSON request as JSON; please check content-type."))
+    }
+  }
```

Comment:
This code exactly is already in `ZioJsonExtractor#parseJson`

We don't want to duplicate such code, maintenance would be harder. Is there an intent with the copy ?

Reply:
It's just that I initially moved that code from the plugins repo, and I didn't realize it was the same :smile:

I reused `ZioJsonExtractor#parseJson` then, it is also more suited that only methods about requests are exposed in `RudderJsonRequest`